src/app/auth.guard.spec.ts

```diff
@@ -1,11 +1,12 @@
 import { TestBed } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
+import { HttpClientModule } from '@angular/common/http';
 
 import { AuthGuard } from './auth.guard';
 
 beforeEach(() => {
   TestBed.configureTestingModule({
-    imports: [RouterTestingModule],
+    imports: [RouterTestingModule, HttpClientModule],
```

Comment:
It is more preferred to use the **HttpClientTestingModule** in the tests
You can read more about it here
[unit-test-and-mock-httpclient](https://www.thecodebuzz.com/angular-unit-test-and-mock-httpclient-get-request/#:~:text=In%20unit%20testing%20its%20important%20to%20mock%20the,services%20that%20make%20HTTP%20calls%20to%20the%20servers.)

Reply:
ok, done